src/api/modules/decrypt/decrypt.background.ts

```diff
@@ -110,18 +114,30 @@ const background: BackgroundModuleFunction<string | Uint8Array> = async (
       false,
       ["decrypt"]
     );
+
     const decrypted = await crypto.subtle.decrypt(options, key, data);
 
     // remove wallet from memory
     freeDecryptedWallet(privateKey);
 
-    return new Uint8Array(decrypted);
+    decryptedData = new Uint8Array(decrypted);
   } else {
     // remove wallet from memory
     freeDecryptedWallet(privateKey);
 
     throw new Error("Invalid options passed", options);
   }
+
+  // request "decrypt" popup
+  await requestUserAuthorization(
```

Comment:
With this change, we would be ALWAYS asking permissions to decrypt data.

This might be an issue but until we rework allowances, I don't think there's a better way. One option could be to do the following check, just like we do with `sign()`:

```
const alwaysAsk = allowance.enabled && allowance.limit.eq(BigNumber("0"));
```

But the issue I have with this is that setting an allowance limit doesn't mention anything to the user about decrypting data (or signing messages).

Reply:
I agree. The copy is not the best for this use cases
Let's go with this approach first, and re-asses later on when allowances has its update